src/sentry/api/endpoints/project_rules.py

```diff
@@ -20,6 +22,18 @@
 from sentry.web.decorators import transaction_start
 
 
+def clean_rule_data(data):
+    for datum in data:
+        if datum.get("name"):
+            del datum["name"]
+
+
+@receiver(pre_save, sender=Rule)
+def pre_save_rule(instance, sender, *args, **kwargs):
+    clean_rule_data(instance.data.get("conditions", []))
+    clean_rule_data(instance.data.get("conditions", []))
```

Comment:
Why do we call this twice?

Reply:
oops I think one was supposed to be "actions"